### PR TITLE
Merge answer/user commenting interface

### LINKED ISSUE #2896: Merge commenting/answering interfaces to reduce redundancy
and make the code more maintainable! This will also bring the new comment emoji system to Questions and Answers.

The basic comment form is here:

https://github.com/publiclab/plots2/blob/master/app/views/comments/_form.html.erb

But Questions have their own style comments. It's cool, but I think we should merge these for consistent UI and easier maintenance:

https://github.com/publiclab/plots2/blob/master/app/views/questions/_comment.html.erb

the form itself here: https://github.com/publiclab/plots2/blob/master/app/views/questions/_form.html.erb

Also, Answers have their own editor. I feel like it could also just be the same editor. But maybe it already does?

https://github.com/publiclab/plots2/blob/master/app/views/questions/_answers.html.erb#L11

There are also various JavaScripts that'll have to be detangled or made consistent, in https://github.com/publiclab/plots2/blob/master/app/assets/javascripts/

Any help appreciated!

### PR DESCRIPTION
**Pull request by ViditChitkara**
fixes #2896 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Ooh, exciting! This will be very cool. 👍

**Comment by jywarren** [comment]
The error is a bit weird -- 

```
 FAIL["test_should_create_answer_comments", Minitest::Result, 119.04303086800002]
 test_should_create_answer_comments#Minitest::Result (119.04s)
        expecting partial <questions/_comment> but action rendered <["comments/_edit", "_edit", "notes/_comment", "_comment"]>.
        Expected {"comments/_edit"=>1, "_edit"=>1, "notes/_comment"=>1, "_comment"=>1} to include "questions/_comment".
        test/functional/comment_controller_test.rb:87:in `block in <class:CommentControllerTest>'
 FAIL["test_should_create_question_comments", Minitest::Result, 119.81190960599997]
 test_should_create_question_comments#Minitest::Result (119.81s)
        expecting partial <questions/_comment> but action rendered <["comments/_edit", "_edit", "notes/_comment", "_comment"]>.
        Expected {"comments/_edit"=>1, "_edit"=>1, "notes/_comment"=>1, "_comment"=>1} to include "questions/_comment".
        test/functional/comment_controller_test.rb:38:in `block in <class:CommentControllerTest>'
{"uid"=>"98740858591", "info"=>{"email"=>"jeff@gmail.com"}}
{"uid"=>"98746858591", "info"=>{"email"=>"bansal.sidharth2990@gmail.com"}}
```

and then:

```

/home/travis/.rvm/gems/ruby-2.4.1/gems/danger-5.6.2/lib/danger/scm_source/git_repo.rb:128:in `find_merge_base': Cannot find a merge base between danger_base and danger_head. (RuntimeError)
```

**Comment by ViditChitkara** [comment]
Hi @jywarren, we need to make changes to the tests since I have changed the partial which renders comments on question page. Tried to reuse the notes/comments partial which would help us rule out the questions/comments partial. Working on that part. 
Yeah, the last error seems wierd.

File: app/views/questions/_answer.html.erb [hunk 1]
```diff
@@ -56,16 +56,16 @@
...
       <a href="#a<%= answer.id %>" class="btn btn-default btn-sm "><i class="fa fa-share-alt"></i></a> &nbsp;
       <a id="answer-<%= answer.id %>-comment-button"  style="color: #006dcc;"><span id="answer-<%= answer.id %>-comment-count"><%= answer.comments.length %></span> Comments <i class="fa fa-caret-down"></i></a>
     </div>
 
-    <div id="answer-<%= answer.id %>-comment" style="margin: 10px; padding:10px; border-radius: 5px; background-color: #eee;">
+    <div id="answer-<%= answer.id %>-comment" style="margin: 10px; padding:10px;display: none;">
       <% answer.comments.each do |comment| %>
-        <%= render partial: "questions/comment", locals: { comment: comment, answer_id: answer.id } %>
+        <%= render partial: "notes/comment", locals: { comment: comment, answer_id: answer.id, is_answer: true } %>
```
Comment (namangupta01) [inline app/views/questions/_answer.html.erb]: @ViditChitkara Shouldn't it be `question/comment` ?

File: app/views/comment/create.js.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-$('#answer-<%= @answer_id %>-comment-section').before('<%= escape_javascript(render :partial => "questions/comment", :locals => {comment: @comment, answer_id: @answer_id}) %>')
+$('#answer-<%= @answer_id %>-comment-section').before('<%= escape_javascript(render :partial => "notes/comment", :locals => {comment: @comment, answer_id: @answer_id}) %>')
```
Comment (namangupta01) [inline app/views/comment/create.js.erb]: Here too. Because if you want to use `notes/comment` then there is no variable `answer_id` in that partial.

Comment (namangupta01) [reply app/views/comment/create.js.erb]: Isn't it? @ViditChitkara

**Comment by ViditChitkara** [comment]
Hi Naman, I'm trying to reuse the notes/comment partial for questions since they are quite similar. This would help us deprecating the question/comments partial. I've made the use of answer_id in notes/comment. With this our comment like feature would be present on question page.

**Comment by ViditChitkara** [comment]
Ideally there should be a single partial for questions and notes, since both of them are derived from "node" in our codebase.

**Comment by namangupta01** [comment]
Cool

**Comment by jywarren** [comment]
It looks like tests on lines 38 and 87 of this file have to be adjusted to account for the template changes: test/functional/comment_controller_test.rb

Then this should be good to go!!!

**Comment by ViditChitkara** [comment]
Cool, doing it now.

**Comment by ViditChitkara** [comment]
Hi @jywarren , I'm done with the changes, the tests are passing now. Regarding codeclimate, it gives "unescaped model attribute error" even though "raw" is used. Saw a similar issue [here](https://github.com/presidentbeef/brakeman/issues/516). It is good to go otherwise!!
Thanks.

**Comment by jywarren** [comment]
Oh this is awesome, thanks. This is for both question comments and answer comments? 

Thanks @ViditChitkara !!!!!

**Comment by ViditChitkara** [comment]
Yes, for both

**Pull request merged**